src/modrover/learner.py

```diff
@@ -1,7 +1,7 @@
 from __future__ import annotations
 
 from operator import attrgetter
-from typing import Callable, Optional
+from typing import Callable, Optional, TYPE_CHECKING
```

Comment:
Interesting! Never see `TYPE_CHECKING` before, is this variable used in anyways?